apps/framework-cli/src/cli/routines.rs

```diff
@@ -585,7 +662,7 @@ pub async fn start_production_mode(
             api_changes_channel,
             metrics.clone(),
             &mut client,
-            &redis_client,
+            &*redis_client.lock().await,
```

Comment:
why does the redis client needs to be locked here?

Reply:
You might not be passing the lock here - since the lock is released when the variable goes out of scope. 
If you want to pass in the locked mutex, you need to pass in the MutexGuard.

Feels like we should minimize the redis client locking to make the client as available as possible